core/src/test/scala/unit/kafka/log/LogCleanerParameterizedIntegrationTest.scala

```diff
@@ -120,12 +120,13 @@ class LogCleanerParameterizedIntegrationTest extends AbstractLogCleanerIntegrati
 
     // Set the last modified time to an old value to force deletion of old segments
     val endOffset = log.logEndOffset
-    log.logSegments.foreach(_.lastModified = time.milliseconds - (2 * retentionMs))
+    log.logSegments.forEach(_.setLastModified(time.milliseconds - (2 * retentionMs)))
     TestUtils.waitUntilTrue(() => log.logStartOffset == endOffset,
       "Timed out waiting for deletion of old segments")
     assertEquals(1, log.numberOfSegments)
 
     cleaner.shutdown()
+    closeLog(log)
```

Comment:
I noticed we did not close the log before reopening it a bit later. It didn't seem to cause any problems, but it made it more difficult to debug test failures (since the behavior is not clearly defined in this case).

Reply:
Good catch!